### PR TITLE
feat/favourite-activity-test-created

### PR DESCRIPTION
**Pull request by ArthurEncr**
# Description

Favorites activity test added to dashboard page tests
- Favorite each page (Seedlots, Create A class and My seedlots) and check if the shortcut is working
- Test highlight and delete the cards
- Test if the empty section is showing

Fixes [#528](https://github.com/bcgov/nr-spar/issues/528)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How to test?

- Export all environment variables (on terminal: export CYPRESS_USERNAME=[USERNAME] and same for password)
- Run the login file test with: yarn `cy:run --spec cypress/e2e/smoke-test/[TEST FILE]` or `npm run cy:open` and select the test

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-544-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-544-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-544-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { FIVE_SECONDS, HALF_SECOND, TWO_SECONDS } from '../../constants';
```
Comment (mgaseta) [inline frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts:1]: You can remove the `FIVE_SECONDS`, it's not being used.

File: frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts [hunk 2]
```diff
@@ -29,4 +30,135 @@ describe('Dashboard page test', () => {
       .find('.recent-activity-subtitle')
       .should('have.text', dashboardPageData.secondSectionSubtitle);
   });
+
+  it('fav activity is working proprely', () => {
```
Comment (mgaseta) [inline frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts:34]: Typo here, `proprely` should be `properly`.

Comment (mgaseta) [inline frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts:34]: IMO this `it` is too big... Can we break it into smaller pieces? For instance, one `it` to make sure the card is working, another to see if the card is correctly appearing, and so on. What do you think? Tagging @RMCampos and @craigyu to get their inputs too.

File: frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts [hunk 3]
```diff
@@ -29,4 +30,135 @@ describe('Dashboard page test', () => {
...
+    // Navigate to Seedlot page
+    cy.navigateTo(NavigationLabels.Seedlots);
+    cy.wait(HALF_SECOND);
+    // Favourite Seedlot page
+    cy.get('.title-favourite')
+      .find('.bx--popover-container')
```
Comment (mgaseta) [inline frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts:40]: Should we make something similar to the theme package and create a global utils for the class prefix? I think it's a good a idea, but the test can become too verbose... What do you think? Tagging @RMCampos and @craigyu too.

File: frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts [hunk 4]
```diff
@@ -29,4 +30,135 @@ describe('Dashboard page test', () => {
...
+      .click();
+    cy.wait(TWO_SECONDS);
+    // Navigate to Dashboard page
+    cy.navigateTo(NavigationLabels.Dashboard);
+    cy.wait(HALF_SECOND);
+    // Check if seedlot card is appearing at fauvorites activities
```
Comment (mgaseta) [inline frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts:46]: Typo on `favourites`.

File: frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts [hunk 5]
```diff
@@ -29,4 +30,135 @@ describe('Dashboard page test', () => {
...
+    cy.get('.title-favourite')
+      .find('.bx--popover-container')
+      .click();
+    cy.wait(TWO_SECONDS);
+
+    // Navigate to Dashboard page
+    cy.navigateTo(NavigationLabels.Dashboard);
+    cy.wait(HALF_SECOND);
+
+    // Check if Create A Class Seedlot card is appearing at fauvorites activities
```
Comment (mgaseta) [inline frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts:69]: Typo on `favourites`.

File: frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts [hunk 6]
```diff
@@ -29,4 +30,135 @@ describe('Dashboard page test', () => {
...
+    cy.get('.title-favourite')
+      .find('.bx--popover-container')
+      .click();
+    cy.wait(TWO_SECONDS);
+
+    // Navigate to Dashboard page
+    cy.navigateTo(NavigationLabels.Dashboard);
+    cy.wait(HALF_SECOND);
+
+    // Check if My Seedlot card is appearing at fauvorites activities
```
Comment (mgaseta) [inline frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts:96]: Typo on `favourites`.

**Review by mgaseta — CHANGES_REQUESTED** [review]
Nice work!

Screenshot from 2023-08-31 12-02-48

Added only some minor markups and started some discussions.

**Comment by craigyu** [comment]
image
Your tests still use `cy.wait()`, please read the description that I tagged you in #532

**Comment by craigyu** [comment]
Please do not bundle test together.
Each add a fav activity and verify the card is showing should be inside its own `it()`, instead of all three bundled together. Same thing for deleting.

**Review by craigyu — CHANGES_REQUESTED** [review]
A test is failing.

**Review by mgaseta — APPROVED** [review]
LGTM! :1st_place_medal: :100: :test_tube:

**Review by craigyu — CHANGES_REQUESTED** [review]

File: frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts [hunk 7]
```diff
@@ -29,4 +30,137 @@ describe('Dashboard page test', () => {
...
+  });
+
+  it('check if delete seedlots favourite card is working', () => {
+    // Delete Seedlots card
+    cy.get('.favourite-activities-cards')
+      .find('.fav-card-main:first')
+      .find('.fav-card-overflow')
+      .click();
+    cy.get(`.${prefix}--overflow-menu-options__option-content:last`)
+      .click();
```
Comment (craigyu) [inline frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts:140]: Two things, this comment applies to the tests below as well:
1. Can we select the button by finding it using text? The current way is clicking on the last option, in the future if we want to add another button, then this test might click on the wrong button.
2. How do we know the deletion worked? I know we are checking if an empty section is rendering, but we should still check if the deleted card is gone. The reason is that if this test passed, we assume that the delete seedlot fav card is working according to the test title, but in actuality we don't know that for sure from this test alone. Let's say we added a third button as mentioned in the first point, and this test is no longer clicking on the delete button, the test will still pass.

**Review by craigyu — CHANGES_REQUESTED** [review]

File: frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts (hunk 7 above)
Comment (craigyu) [reply frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts:140]: btw I apologize for all the trickled in request changes 😬

Comment (ArthurEncr) [reply frontend/cypress/e2e/smoke-test/dashboard-page.cy.ts]: 1 - Very valid point, I made the change

2 - I added one more check to see if the number of cards is changing ;)

**Review by craigyu — APPROVED** [review]
Great job Arthur!

**Pull request merged**